frontend/src/hooks/secret-operations/useCopySecretToClipboard.ts

```diff
@@ -0,0 +1,53 @@
+import { useCallback, useEffect } from "react";
+
+import { createNotification } from "@app/components/notifications";
+
+import { useFetchSecretValue } from "../api/dashboard/queries";
+import { TGetSecretValueDTO } from "../api/dashboard/types";
+import { useToggle } from "../useToggle";
+
+interface CopySecretHookParams {
+  getFetchedValue?: () => string;
+  fetchSecretParams: TGetSecretValueDTO;
+}
+
+export function useCopySecretToClipBoard({
```

Comment:
A lot of component that need to copy the actual value of the secret, each implemented it's own logic, made sense to pull it here.